[15.0] [MIG] `website_sale_cart_expire`

ivantodorovich opened the pull request:
Standard migration plus a few extra feature and fixes, each in a separate commit to ease review and/or backporting

ivantodorovich committed:
[ADD] website_sale_cart_expire

Module that allows to automatically cancel carts without
activity after a configurable time.

oca-travis committed:
[UPD] Update website_sale_cart_expire.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

ioans73 committed:
[MIG] website_sale_cart_expire: Migration to 14.0

oca-travis committed:
[UPD] Update website_sale_cart_expire.pot

OCA-git-bot committed:
[UPD] README.rst

ivantodorovich committed:
[MIG] website_sale_cart_expire: Migration to 15.0

petrus-v reviewed (approved):
Thanks for this migration !

Code review looks good to me.

ivantodorovich committed:
[FIX] website_sale_cart_expire: Don't cancel sent quotations

In a typical website flow, quotations are moved to sent state when
a payment by Wire transfer is chosen.

In this scenario, the quotation shouldn't be automatically cancelled.

It's up to the salesman to determine when a sent quotation is expired,
as technically speaking this isn't just a cart anymore.

ivantodorovich committed:
[FIX] website_sale_cart_expire: Don't cancel carts in payment

ivantodorovich committed:
[FIX] Use savepoint and autocommit to avoid possible cron crash

ivantodorovich committed:
[IMP] website_sale_cart_expire: Display the cart expiration timer on website

petrus-v commented on website_sale_cart_expire/models/website.py:
```diff
@@ -0,0 +1,61 @@
+            # We don't want to cancel carts that are already in payment.
+            "|",
+            ("transaction_ids", "=", False),
+            "!",
+            ("transaction_ids.state", "in", ["pending", "authorized", "done"]),
```
It's common to see the negative term `"!"` used directly as it's translated by odoo from operator, so I would suggest but won't lock merging process for that 

```suggestion
            ("transaction_ids.state", "not in", ["pending", "authorized", "done"]),
```

ivantodorovich replied:
In this case it's not the same, though.

Using the `!` operator it means: _Orders that don't have any transaction in these states_
Using the `not in` operator it means: _Orders having transactions in different states than these_

ioans73 reviewed (approved):
LGTM 👍🏻

ivantodorovich commented:
/ocabot merge nobump
